cinema/serializers.py

```diff
@@ -1,24 +1,83 @@
 from rest_framework import serializers
 
-from cinema.models import Movie
+from cinema.models import Movie, Actor, Genre, CinemaHall
 
 
-class MovieSerializer(serializers.Serializer):
-    id = serializers.IntegerField(read_only=True)
-    title = serializers.CharField(max_length=255)
-    description = serializers.CharField()
-    duration = serializers.IntegerField()
+class MovieSerializer(serializers.ModelSerializer):
+    actors = serializers.PrimaryKeyRelatedField(many=True,
+                                                queryset=Actor.objects.all())
+    genres = serializers.PrimaryKeyRelatedField(many=True,
+                                                queryset=Genre.objects.all())
+
+    class Meta:
+        model = Movie
+        fields = ['id', 'title', "duration", "actors", "genres"]
 
     def create(self, validated_data):
-        return Movie.objects.create(**validated_data)
+        actors = validated_data.pop("actors")
+        genres = validated_data.pop("genres")
+        movies = Movie.objects.create(**validated_data)
+        movies.set(actors)
+        movies.set(genres)
```

Comment:
There is an error in the create method. The correct method to add ManyToMany relationships is to use 'set' on the related manager, e.g., 'movies.actors.set(actors)' and 'movies.genres.set(genres)'. The current code attempts to call 'set' directly on the Movie instance, which is incorrect.